src/battle/battle_script_commands.c

```diff
@@ -1636,6 +1707,9 @@ u32 CalculateBallShakes(void *bw, struct BattleStruct *sp)
         if (criticalCapture) // succeeded the one chance it had
             i = i | 0x80; // change the flow of the ball callback to make sure that critical captures only shake once then succeed.  if it shakes, it succeeds, though
 
+        if(sp->item_work = ITEM_FRIEND_BALL) // To catch a 1 shake capture
+            SetMonData(Battle_GetClientPartyMon(bw,sp->defence_client,0),ID_PARA_friend,200);
+            
```

Comment:
equivalence checks in C are `==`

additionally, the 3rd arg for `SetMonData` needs to be a pointer to the data.  not 100% on why GF handles it that way, but hey:
```c
if (sp->item_work == ITEM_FRIEND_BALL)
{
    u32 friendship = 200;
    SetMonData(Battle_GetClientPartyMon(bw,sp->defence_client,0), ID_PARA_friend, &friendship);
}
```

Reply:
also i don't think this is the right place to be inserting this--should only be inserting it for success conditions.  critical captures are not 100% successful

perhaps insert it above the check immediately above with the `criticalCapture` and check if `i == caughtMons`:


https://github.com/BluRosie/hg-engine/blob/b7aa067f2a7e35bacc596b8db3a00696c6105cb9/src/battle/battle_script_commands.c#L1710-L1711

```c
if (sp->item_work == ITEM_FRIEND_BALL && i == caughtMons)
{
    u32 friendship = 200;
    SetMonData(Battle_GetClientPartyMon(bw,sp->defence_client,0), ID_PARA_friend, &friendship);
}
```

also still need to insert it here https://github.com/BluRosie/hg-engine/blob/b7aa067f2a7e35bacc596b8db3a00696c6105cb9/src/battle/battle_script_commands.c#L1660-L1661

Reply:
Not sure why but I was processing the break statements in the for loop about as returns, thinking by the time it got the if statement the throw was successful. I'll check i against caughtMons.